lib/chef/resource/sudo.rb

```diff
@@ -185,7 +185,7 @@ def platform_config_prefix
                       setenv:             new_resource.setenv,
                       env_keep_add:       new_resource.env_keep_add,
                       env_keep_subtract:  new_resource.env_keep_subtract
-            verify "#{new_resource.visudo_binary} -cf %{path}" if visudo_present?
+            verify "cat #{new_resource.config_prefix}/sudoers %{path} | #{new_resource.visudo_binary} -cf -" if visudo_present?
```

Comment:
```suggestion
             verify " #{new_resource.visudo_binary} -c" if visudo_present?
```

Reply:
I did the suggested change and created file `foo` in sudoers.d/ with below content.
```bash
Cmnd_Alias TEST = /usr/bin/ls, /usr/bin/cat
admin ALL=(ALL) ALL
```
Now, When I am running cookbook with `sudo` resource which have the same `Cmnd_Alias`, it runs successfully, but after that, it breaks the `sudo` command
```bash
[vagrant@localhost chef]$ sudo -s
>>> /etc/sudoers.d/foo: Alias "TEST" already defined near line 4 <<<
sudo: parse error in /etc/sudoers.d/foo near line 4
sudo: no valid sudoers sources found, quitting
sudo: unable to initialize policy plugin
```
So, we have to merge the content of `/etc/sudoers` file and the current .d file using 
```ruby
cat #{new_resource.config_prefix}/sudoers %{path} | #{new_resource.visudo_binary} -cf -" if visudo_present?
```